[interp] support ARM soft float ABI for native transitions

## Steps to Reproduce

1. Build the following Main.
2. Run with arm soft float under the interpreter


```
        static void Main(string[] args)
        {
            TimeSpan timeSpan = TimeSpan.FromMilliseconds(548);
            Console.WriteLine(timeSpan);      
        }
```

### Current Behavior

Divide by Zero exception in the TimeSpan print.

### Expected Behavior

The print to occur successfully.

## On which platforms did you notice this

[ ] macOS
[x] Linux
[ ] Windows

**Version Used**:

```
Mono JIT compiler version 5.12.0.226 (tarball Mon May 13 17:28:43 UTC 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       normal
        Notifications: epoll
        Architecture:  armel,vfp
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          supported, not enabled.
        GC:            sgen (concurrent by default)
```

## Stacktrace

```
Unhandled Exception:
System.DivideByZeroException: Attempted to divide by zero.
  at System.Globalization.TimeSpanFormat.FormatStandard (System.TimeSpan value, System.Boolean isInvariant, System.String format, System.Globalization.TimeSpanFormat+Pattern pattern) <0x1d1ef30 + 0x001a8> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at System.Globalization.TimeSpanFormat.Format (System.TimeSpan value, System.String format, System.IFormatProvider formatProvider) <0x1d1e8b0 + 0x00084> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at System.TimeSpan.ToString (System.String format, System.IFormatProvider formatProvider) <0x1d169a8 + 0x0003e> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at System.IO.TextWriter.WriteLine (System.Object value) <0x1d16758 + 0x0003e> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at System.IO.TextWriter+SyncTextWriter.WriteLine (System.Object value) <0x1d16618 + 0x00014> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at (wrapper synchronized) System.IO.TextWriter+SyncTextWriter.WriteLine(object)
  at System.Console.WriteLine (System.Object value) <0x1d16420 + 0x0000c> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at rioapp.Program.Main (System.String[] args) <0x1cac2e8 + 0x00032> in <f458eecb9e34478f8569bde63e909e97>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.DivideByZeroException: Attempted to divide by zero.
  at System.Globalization.TimeSpanFormat.FormatStandard (System.TimeSpan value, System.Boolean isInvariant, System.String format, System.Globalization.TimeSpanFormat+Pattern pattern) <0x1d1ef30 + 0x001a8> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at System.Globalization.TimeSpanFormat.Format (System.TimeSpan value, System.String format, System.IFormatProvider formatProvider) <0x1d1e8b0 + 0x00084> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at System.TimeSpan.ToString (System.String format, System.IFormatProvider formatProvider) <0x1d169a8 + 0x0003e> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at System.IO.TextWriter.WriteLine (System.Object value) <0x1d16758 + 0x0003e> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at System.IO.TextWriter+SyncTextWriter.WriteLine (System.Object value) <0x1d16618 + 0x00014> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at (wrapper synchronized) System.IO.TextWriter+SyncTextWriter.WriteLine(object)
  at System.Console.WriteLine (System.Object value) <0x1d16420 + 0x0000c> in <be1e924bc64546e1a45caab4fa78ec49>:0
  at rioapp.Program.Main (System.String[] args) <0x1cac2e8 + 0x00032> in <f458eecb9e34478f8569bde63e909e97>:0
```

--- a/mono/mini/mini-arm.c
+++ b/mono/mini/mini-arm.c
@@ -1637,7 +1637,10 @@ arg_get_storage (CallContext *ccontext, ArgInfo *ainfo)
 			return &ccontext->gregs [ainfo->reg];
 		case RegTypeHFA:
 		case RegTypeFP:
-			return &ccontext->fregs [ainfo->reg];
+			if (IS_HARD_FLOAT)
+				return &ccontext->fregs [ainfo->reg];
+			else
+				return &ccontext->gregs [ainfo->reg];
 		case RegTypeBase:
 			return ccontext->stack + ainfo->offset;
 		default:

